tools/bin/cdk-auto-release/update-changelog.sh

```diff
@@ -0,0 +1,413 @@
+      return 0 # Not an error, just nothing to commit
+    fi
+    if ! git commit -m "$commit_message"; then
+        error_exit "Failed to commit changes for ${folder_name}"
+    fi
+
+    log_info "Pushing changes to origin..."
+    if ! git push; then
+        # Consider adding retry logic or better error handling for push failures
+        error_exit "Failed to push changes for ${folder_name}"
+    fi
+    log_info "Successfully committed and pushed changes for ${folder_name}"
+}
+
+
+# --- Main Changelog Update Logic for a Single Connector ---
+
+# Function to update the changelog for a specific connector
+# Usage: update_single_connector_changelog <pr_number> "connector_type-connector_name" "Update Message"
+update_single_connector_changelog() {
+    local pr_number="$1"
+    local full_connector_name="$2"
+    local message="$3"
+
+    # --- Add this debug line ---
+    echo "DEBUG: Value received inside function pr_number=[${pr_number}]" >&2
+    # --- End of added line ---
+
+    # Validate inputs (basic checks)
+    if ! [[ "$pr_number" =~ ^[0-9]+$ ]]; then
+        error_exit "Invalid PR number '${pr_number}' passed to update_single_connector_changelog."
+    fi
+    if [ -z "$full_connector_name" ] || [ -z "$message" ]; then
+        error_exit "Connector name and message must be provided to update_single_connector_changelog."
+    fi
+
+    log_info "--- Starting changelog update for ${full_connector_name} ---"
+
+    local filepath separator_line_num data_line_num latest_version new_version new_row
+
+    filepath=$(resolve_connector_file "$full_connector_name")
+    separator_line_num=$(find_changelog_separator_line "$filepath")
+    data_line_num=$((separator_line_num + 1))
+    latest_version=$(get_latest_version "$filepath" "$data_line_num")
+    new_version=$(increment_version "$latest_version")
+    new_row=$(prepare_new_changelog_row "$new_version" "$pr_number" "$message")
+    insert_changelog_row "$filepath" "$separator_line_num" "$new_row"
+
+    log_info "--- Completed changelog update for ${full_connector_name} ---"
+    echo "$filepath" # Return the path of the modified file for git operations
+}
+
+
+# --- Main Execution ---
+main() {
+    # Enable stricter error handling
+    # set -e exits immediately if a command exits with a non-zero status.
+    # set -o pipefail causes a pipeline to return the exit status of the last command
+    # that exited with a non-zero status, or zero if no command exited with a non-zero status.
+    set -eo pipefail
+
+    local branch_name="$1" # Expect branch name as the first argument
+    validate_script_args "$@" # Validate we got the branch name
+
+    fetch_updates
+    local merge_base
+    merge_base=$(find_merge_base)
+
+    # Pull potentially new changes fetched from master (optional, fetch might be enough)
+    log_info "Pulling latest changes..."
+    if ! git pull; then
+       log_info "Warning: git pull failed, continuing with fetched state."
+    fi
```

Comment:
Clarification: why do we have to do this? Is it in case someone has merged something into master while the command is running?

Reply:
Not from master, it is pull to get the commit which bump the version. I need to change the commit. I am not sure why the checkout command is not enough.